### PR TITLE
devolearn/lineage_population_model: add postprocessing algorithm

### PR DESCRIPTION
**Pull request by Mainakdeb**
* The new post processing algorithm modifies outputs from the lineage population model. 
* The attached plots showcase postprocessed and non-postprocessed outputs (input video from [EPIC dataset](https://epic.gs.washington.edu/), [id: 20080709_ceh-27_1_L1](https://epic.gs.washington.edu/expts/20080709_ceh-27_1_L1.html) )
* The loss is approximately 30% lower compared to non post processed outputs (MSELoss over 36 videos)
* The dotted lines represent model outputs, and the non dotted lines represent true values.

non_postprocess_nb
postprocess_nb

### CODE REVIEW

File: devolearn/lineage_population_model/lineage_population_model.py [hunk 1]
```diff
@@ -172,7 +182,7 @@ def predict_from_video(self, video_path, csv_name  = "foo.csv", save_csv = False
 
 
         
-    def create_population_plot_from_video(self, video_path, save_plot = False, plot_name = "plot.png", ignore_first_n_frames = 0, ignore_last_n_frames = 0, notebook_mode = False):
+    def create_population_plot_from_video(self, video_path, save_plot = False, plot_name = "plot.png", ignore_first_n_frames = 0, ignore_last_n_frames = 0, notebook_mode = False, postprocess=False):
         """Plots all the predictions from a video into a matplotlib.pyplot
```
Comment (Mayukhdeb) [inline devolearn/lineage_population_model/lineage_population_model.py:191]: You did not update the docstring for the new argument, so please do that

**Comment by ABD-01** [comment]
Hi!
I think the function `postprocess_preds` function can be improved further by using boolean indices like this : 
```py
def postprocess_preds(self, preds):
    """
    Args:
        preds (numpy.ndarray): array that needs to be postprocessed.
    """
    for i in range(1, len(preds)):
        mask = preds[i] <= preds[i-1]
        preds[i][mask] = preds[i-1][mask] 

    return(preds)
```
Analysis (on 1000x1000 np.ndarray):
image

**Comment by Mayukhdeb** [comment]
Interesting, can you provide us with the benchmarks which shows the speedup in `devolearn` ? You can make the modifications in your own fork and test it out on colab. 

cc: @Mainakdeb

**Comment by Mainakdeb** [comment]
> Interesting, can you provide us with the benchmarks which shows the speedup in `devolearn` ? You can make the modifications in your own fork and test it out on colab.
> 
> cc: @Mainakdeb

Hi, I tried out the new post-processing function suggested by @ABD-01 on a sample video. Seems like this technique works well for higher dimensional numpy ndarays, but the difference is barely noticeable in the arrays we are working with.

Screenshot from 2021-04-06 02-19-08

Screenshot from 2021-04-06 02-19-28

cc: @Mayukhdeb

**Pull request merged**